File: admision24-1/RetoGPC/retoGPC.py
# def pedir_valores():
#     a = int(input())
#     b = int(input())
#     x = int(input())
#     if (a < 1 or a > 6) or (b < 1 or b > 6) or (x < 20 or x > 1000):
#         pedir_valores()
#
#     return a, b, x
#
#
# a, b, x = pedir_valores()
#
# print(a, b, x)

a = int(input())

b = int(input())

x = int(input())

serie = 0
encontrado = False

for i in range(x):
    serie += a
    if serie == x:
        encontrado = True
        break
    serie += b
    if serie == x:
        encontrado = True
        break

if encontrado:
    print("usa botas")
else:
    print("usa zapatillas")

File: admision24-1/RetoGPC/retoGPC.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int a, b,x;
    do {
        cin >> a >> b >> x;
    }while ((a < 1 && a > 6 && b < 1 && b > 6) && (20 > x && x >= 100));

    int serie = 0;
    vector<int> numerosSerie;
    bool numeroEncontrado = false;
    int contador = 0;

    for (int i = 0; i <= x; i++) {
        serie += a;
        numerosSerie.push_back(serie);
        serie += b;
        numerosSerie.push_back(serie);
        if (numerosSerie[i] <= x) {
            contador++;
        }
        else {
            break;
        }
    }

    for (int i = 0; i <= contador; i++) {
        if (numerosSerie[i] == x) {
            numeroEncontrado = true;
        }
    }

    if (numeroEncontrado) {
        cout << "usa botas";
    }
    else {
        cout << "usa zapatillas";
    }

    return 0;
}